keras/engine/training.py

```diff
@@ -259,6 +260,12 @@ def compile(self, optimizer,
                         weight = K.placeholder(ndim=2,
                                                name=name + '_sample_weights')
                         sample_weight_modes.append('temporal')
+                    elif sample_weight_mode.get(name) == 'element':
+                        ndim = len(self.output_shape[i]) - 1 if isinstance(self.output_shape, list)\
+                            else len(self.output_shape) - 1
```

Comment:
Avoid avoid many logical statements on a single line, which harms readability. Instead, use a if/else block.

Reply:
Hi @fchollet . I think that kind of syntax was requested by @pavithrasv in the discussion from the review on 8 Aug that you can find above. Did I misunderstand something?

Reply:
Using if/else here definitely improves readability. In the previous discussion I had request to de-dup the code for creating placeholder which you have done already.

Reply:
No problem, I also like it better. I was just trying to adapt to whatever style you preferred. Now implemented.